tools/nhlt-parse.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2020 Intel Corporation. All rights reserved.
+#
+# usage: nhlt-parse.py <nhtl-dump>
+#
+# examples:
+#   cat /sys/firmware/acpi/tables/NHLT >nhtl.bin
+#   ./nhtl-parse.py nhtl.bin
+#
+# references:
+#  - https://01.org/sites/default/files/595976_intel_sst_nhlt.pdf
+
+from collections import namedtuple
+import struct
+import sys
+
+if len(sys.argv) == 1:
+    infile = '/sys/firmware/acpi/tables/NHLT'
+else:
+    infile = sys.argv[1]
+
+f = open(infile, 'rb')
+
+record = f.read()
+
+Link = namedtuple('Link', 'DescriptorLength LinkType InstanceID VendorID DeviceID RevID SubsystemID DevType Direction VirtualBusId')
+
+link_types = { 0:'HDA', 1:'DSP', 2:'PDM', 3:'SSP', 4:'Reserved4', 5:'SoundWire' }
+
+# skip ACPI table header
+p = 36
+n_endpoints = struct.unpack('<B', record[p:p+1])
+
+p = p+1
+
+print('NHLT %u endpoints' % n_endpoints)
+
+dmic_count = 0
+
+for i in range(0,n_endpoints[0]):
+    # ENDPOINT_DESCRIPTOR
+    ffmt = '<IBBHHHIBBB'
+    fsize = struct.calcsize(ffmt)
+    endpoint_descriptor = record[p:p+fsize]
+    link = Link._make(struct.unpack(ffmt, endpoint_descriptor))
+    link_type = link._asdict()['LinkType']
+    print(link)
+    print('NHLT link type %s' % link_types[link_type])
+    if link_type == 2:
+        dmic_count += 1
```

Comment:
this doesn't seem to work with the attachment in https://bugzilla.kernel.org/attachment.cgi?id=293599

We should be getting an invalid config and we get

````
NHLT 1 endpoints
Link(DescriptorLength=3054, LinkType=2, InstanceID=0, VendorID=32902, DeviceID=44576, RevID=4332, SubsystemID=270864439, DevType=1, Direction=1, VirtualBusId=0)
NHLT link type PDM
NHLT SPECIFIC_CONFIG size 2
NHLT FORMAT CONFIGS 1
NHTL: 1 dmics described
```